### PR TITLE
fix serialization of bounding_box with quantities

### PR DESCRIPTION
**Pull request by nden**
Fixes issue brought up in #8037.

### CODE REVIEW

**Comment by pllim** [comment]
Does this close or complement #8037? Does this need a test?

**Comment by nden** [comment]
@pllim It replaces #8037. It has a test.

File: astropy/io/misc/asdf/tags/transform/basic.py [hunk 1]
```diff
@@ -28,8 +28,8 @@ def _from_tree_base_transform_members(cls, model, node, ctx):
         # TODO: Remove domain in a later version.
         if 'domain' in node:
             model.bounding_box = cls._domain_to_bounding_box(node['domain'])
-        elif 'bounding_box' in node:
-            model.bounding_box = node['bounding_box']
+        if 'bounding_box' in node:
```
Comment (astrofrog) [inline astropy/io/misc/asdf/tags/transform/basic.py]: Why the change from ``elif`` to ``if``? If you want the 'bounding_box' clause to take precedence, maybe put it first and then use an ``elif``?

Comment (nden) [reply astropy/io/misc/asdf/tags/transform/basic.py]: I was considering removing `domain`. It was replaced by `bounding_box` quite a while ago.
@drdavella Is there any reason NOT to remove `domain`? I believe it was only used by jwst and by now there shouldn't be any files left with it.

Comment (drdavella) [reply astropy/io/misc/asdf/tags/transform/basic.py]: @nden it's okay with me to remove. It was removed in v1.1.0 of the Standard, so there's no requirement to support backwards compatibility here given the version of `TransformType` is already pinned to 1.1.0.

Comment (nden) [reply astropy/io/misc/asdf/tags/transform/basic.py]: After more conversations locally - removing it now doesn't appear to be a problem.

**Comment by bsipocz** [comment]
Comment has been addressed, so I'm merging this one, too.

**Comment by bsipocz** [comment]
Thank you @nden!
